internal/provider/resource_ssh_key.go

```diff
@@ -160,91 +163,39 @@ func (s *sshKeyResource) Read(ctx context.Context, req resource.ReadRequest, res
 		return
 	}
 
-	var plan sshKeyResourceModel
+	var state sshKeyResourceModel
 
 	// Read Terraform configuration from the request into the model
-	resp.Diagnostics.Append(req.State.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	modelName, keyIdentifier := retrieveModelKeyNameFromID(plan.ID.ValueString(), &resp.Diagnostics)
+	modelName, identifier := retrieveModelKeyNameFromID(state.ID.ValueString(), &resp.Diagnostics)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
 	result, err := s.client.SSHKeys.ReadSSHKey(&juju.ReadSSHKeyInput{
 		Username:      s.client.Username(),
 		ModelName:     modelName,
-		KeyIdentifier: keyIdentifier,
+		KeyIdentifier: identifier,
 	})
 	if err != nil {
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read ssh key, got error: %s", err))
 		return
 	}
-	s.trace(fmt.Sprintf("read ssh key resource %q", plan.ID.ValueString()))
+	s.trace(fmt.Sprintf("read ssh key resource %q", state.ID.ValueString()))
 
-	plan.ModelName = types.StringValue(result.ModelName)
-	plan.Payload = types.StringValue(result.Payload)
+	state.ModelName = types.StringValue(result.ModelName)
+	state.Payload = types.StringValue(result.Payload)
 
 	// Set the plan onto the Terraform state
-	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
 }
 
 func (s *sshKeyResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	// Prevent panic if the provider has not been configured.
```

Comment:
Nice, this is also in `v1`. We'll need to merge `main` into `v1` to keep it up to date.